winlogbeat/eventlog/eventlog_windows.go

```diff
@@ -407,3 +405,40 @@ func queryEventMessageFiles(providerName, sourceName string) ([]Handle, error) {
 		sourceName)
 	return handles, nil
 }
+
+// readRetryErrorHandler handles errors returned from the readEventLog function
+// by attempting to correct the error through closing and reopening the event
+// log.
```

Comment:
Closing / Reopening to fix an issue sounds a little bit like a hack :-) But it seems to work.